### ISSUE TITLE
application that can benefit from TV

### ISSUE THREAD

**Issue by unxed** [magiblot/tvision#24]
> If you know of any Turbo Vision applications whose source code has not been lost and that could benefit from this, let me know.

https://en.wikipedia.org/wiki/DOS_Navigator

> source code has not been lost

Original DOS version:
https://www.ritlabs.com/download/dn/dn151src.zip

Win32 & os/2 adoption tryouts
http://www.dnosp.com/files/dn2/dn2s214.rar

But that's Pascal. Can it make friends with TV in C?

**Comment by unxed** [comment]
I even [managed to build it in Linux](https://github.com/unxed/dn2l/tree/first_try), but it is written for ancient Virtual Pascal that seems to have some bug in it's RTL preventing any app (including simple helloworld) from running :(

**Comment by magiblot** [comment]
Hi unxed!

> But that's Pascal. Can it make friends with TV in C?

Yes, after rewriting everything in C++ :). This can be a good exercise for anyone wanting to learn either Pascal or C++.

> I even managed to build it in Linux, but it is written for ancient Virtual Pascal that seems to have some bug in it's RTL preventing any app (including simple helloworld) from running :(

It seems to contain the whole Turbo Vision source code, so even if it compiles, it won't run as it's probably doing DOS or Windows syscalls.

**Comment by magiblot** [comment]
By the way, I have seen your post at https://habr.com/en/post/524370/. Look at what you have done to my repositories :)

<details>
<summary>turbo</summary>

![Screenshot_20201023 magiblot turbo](https://user-images.githubusercontent.com/20713561/96947006-3cca7e00-14e2-11eb-99d1-859cf1fce9d8.png)
</details>

<details>
<summary>tvision</summary>

![Screenshot_20201023 magiblot tvision](https://user-images.githubusercontent.com/20713561/96947008-3dfbab00-14e2-11eb-9c91-834b78e7c33d.png)
</details>

**Comment by unxed** [comment]
> By the way, I have seen your post at https://habr.com/en/post/524370/. Look at what you have done to my repositories :)
> turbo
> tvision

Cool! Well, still, a crowd of fans of the good old TUI gathered in the comments, and this is the largest site in Russia about IT.

**Comment by IngvarRiga** [comment]
Hi!
see that: http://www.sigala.it/sergio/tvision/index.html

**Comment by IngvarRiga** [comment]
See that? 
http://www.sigala.it/sergio/tvision/resources.html

**Comment by magiblot** [comment]
Hi @IngvarRiga! Yes, I have seen that page many times (and even ported some of its apps, e.g. https://github.com/magiblot/tvision/discussions/54). But most of these are just small examples.

**Comment by IngvarRiga** [comment]
Another hi ! ))
see to russian resources (pascal, but.... ): https://pascal.sources.ru/tvision/index.htm

pay attention to the last project in the list - it is a builder of screen forms as in Delphi. I wrote this myself and also in Pascal, but too long ago and the code has not been preserved. (https://pascal.sources.ru/tvision/resrs2.htm)


... And thank you for your wonderful work! Now I'll just sort out the idiotic CMake and start writing my own. I can't set up the project for compilation in any way.

**Comment by magiblot** [comment]
> Now I'll just sort out the idiotic CMake and start writing my own. I can't set up the project for compilation in any way.

What errors do you get? If you want you can open a new issue and share the details there

**Comment by IngvarRiga** [comment]
No, I have to figure it out myself, thanks for the offer. The fact is that I have assembled your project, but I can't also assemble my own, with the library connected. I am used to using normal C++ IDEs like MSVC and the like and have almost never used assemblies via make, so it is unexpected for me when I cannot identify a problem, for example, not a certain symbol during assembly or not given any compilation option of the project. In CMake there are no explicit indications of options, etc. Well... I probably haven't figured out what to do yet. So I'm figuring it out. For example, now when linking, I get a message that the linker cannot find the virtual method table. I'm in shock.... :)

**Comment by magiblot** [comment]
I would not like you to have a bad experience with CMake, so if you want a hint, this comment contains a possible solution: https://github.com/magiblot/tvision/issues/50#issuecomment-808296503.

**Comment by IngvarRiga** [comment]
TV Dialog Design... no src. http://www.pbear.com/ but its free & work

**Comment by unxed** [comment]
Adopted Free Vision (Free Pascal's Turbo Vision port) to UTF-8 also. Still work in progress, although test app is already able to show interface properly, open utf-8 encoded text file with utf-8 encoded name and display it correctly.

https://github.com/unxed/fv_utf8

**Comment by magiblot** [comment]
That's good news! Good job!

**Comment by unxed** [comment]
Oh, there is also [official attempt](https://gitlab.com/freepascal.org/fpc/source/-/issues/21373#note_833019965) of adopting Free Pascal's Free Vision to unicode, awaiting to be integrated to main branch. I don't knew that.

**Comment by unxed** [comment]
Unicode Free Vision (Free Pascal's Turbo Vision implementation) has been merged to Free Pascal main branch: https://github.com/fpc/FPCSource/tree/main/packages/fv/src

(see units with names starting with "u")

**Comment by okbob** [comment]
so 12. 2. 2022 v 15:54 odesílatel unxed ***@***.***> napsal:

> Unicode Free Vision (Free Pascal's Turbo Vision implementation) has been
> merged to Free Pascal main branch:
> https://github.com/fpc/FPCSource/tree/main/packages/fv/src
>
> (see units with names starting with "u")
>

great news

Thank you for info

Pavel
